pkg/controller/master/setting/rke2_registries.go

```diff
@@ -0,0 +1,129 @@
+package setting
+
+import (
+	"encoding/json"
+	"reflect"
+
+	"github.com/rancher/wharfie/pkg/registries"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"sigs.k8s.io/kustomize/kyaml/yaml"
+
+	harvesterv1 "github.com/harvester/harvester/pkg/apis/harvesterhci.io/v1beta1"
+	"github.com/harvester/harvester/pkg/util"
+)
+
+func (h *Handler) syncContainerdRegistry(setting *harvesterv1.Setting) error {
+	registryFromSetting := &registries.Registry{}
+	if err := json.Unmarshal([]byte(setting.Value), registryFromSetting); err != nil {
+		return err
+	}
+
+	registryFromSettingYaml, err := yaml.Marshal(registryFromSetting)
+	if err != nil {
+		return err
+	}
+
+	secret, err := h.secretCache.Get(util.CattleSystemNamespaceName, util.ContainerdRegistrySecretName)
+	if errors.IsNotFound(err) {
+		newSecret := &corev1.Secret{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      util.ContainerdRegistrySecretName,
+				Namespace: util.CattleSystemNamespaceName,
+			},
+			Data: map[string][]byte{
+				util.ContainerdRegistryFileName: registryFromSettingYaml,
+			},
+		}
+
+		if _, err := h.secrets.Create(newSecret); err != nil {
+			return err
+		}
+		return nil
+	} else if err != nil {
+		return err
+	}
+
+	isStale, err := h.isStaleSecret(secret, registryFromSetting)
```

Comment:
can we try to compare the setting values and secret data before the update to skip the custom compares?

Reply:
I think we can't do it, because we remove credentials in the setting, so their string value will not be the same.